README.md

```diff
@@ -86,6 +86,169 @@ server.enableAuth('user', 'pass');
+serverHandler = server.listen(8088, 'localhost');
+
+client = rpc.Client.$create(8088, 'localhost');
+client.basicAuth('user', 'pass');
+
+var params = ['Hello!, Authorization!'];
+client.call('echo', params, function (err, result) {
+    if (err) { /* do something with error */ }
+    // do something with result
+    console.log(result); // in this case, 'Hello, Authorization!'
+});
+
+serverHandler.close();
+server = null;
+client = null;
+```
+
+Cookie Authorization Communication
+
+```js
+var rpc = require('json-rpc2');
+
+var server = rpc.Server.$create({
+    'websocket': true
+});
+
+server.define('echo', function (opts, args, callback) {
+    // function callback(error, result)
+    callback(null, args[0]);
+});
+
+server.enableCookieAuth(function (cookieValue) {
+    // do call to user service or pass it to middleware
+    return (cookieValue === 'validCookieValue');
+});
+serverHandler = server.listen(8088, 'localhost');
+
+client = rpc.Client.$create(8088, 'localhost');
+client.cookieAuth('validCookieValue');
+
+var params = ['Hello!, Authorization!'];
+client.call('echo', params, function (err, result) {
+    if (err) { /* do something with error */ }
+    // do something with result
+    console.log(result); // in this case, 'Hello, Authorization!'
+});
+
+serverHandler.close();
+server = null;
+client = null;
+```
+
+Cookie Authorization Communication
+
+```js
+var rpc = require('json-rpc2');
+
+var server = rpc.Server.$create({
+    'websocket': true
+});
+
+server.define('echo', function (opts, args, callback) {
+    // function callback(error, result)
+    callback(null, args[0]);
+});
+
+server.enableJWTAuth(function (tokenValue) {
+    // do call to user service or pass it to middleware
+    return (cookieValue === 'validCookieValue');
+});
+serverHandler = server.listen(8088, 'localhost');
+
+client = rpc.Client.$create(8088, 'localhost');
+client.jwtAuth('validTokenValue');
+
+var params = ['Hello!, Authorization!'];
+client.call('echo', params, function (err, result) {
+    if (err) { /* do something with error */ }
+
+    // do something with result
+    console.log(result); // in this case, 'Hello, Authorization!'
+});
+
+serverHandler.close();
+server = null;
+client = null;
+```
+
+Switching between Authorization Methods
+
+```js
+var rpc = require('json-rpc2');
+
+var server = rpc.Server.$create({
+    'websocket': true
+});
+
+server.define('echo', function (opts, args, callback) {
+    // function callback(error, result)
+    callback(null, args[0]);
+});
+
+server.listen(8088, 'localhost');
+
+// Server Authorization
+server.enableBasicAuth('user', 'pass');
+server.enableCookieAuth(function (cookieValue) {
+    // handle cookie validity through a user service or middleware
+    return (cookieValue === 'validCookieValue');
+});
+server.enableJWTAuth(function (tokenValue) {
+    // handle JWT token validity through a user service or middleware
+    return (tokenValue === 'validTokenValue');

```

Comment:
when the validation routine is async, how it goes? maybe we could add promises (since you mentioned in #26). 

could use `Bluebird.try` that can resolve either the return value of the return of a promise, it's also backwards compatible